packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx

```diff
@@ -0,0 +1,359 @@
+        let distance: number = this.getMaxScrollDistance(availableWidth, itemWidths) + availableWidth;
+
+        for (let i: number = itemWidths.length - 1; i >= 0; i--) {
+            if (distance - itemWidths[i] < distanceFromBeginning - availableWidth) {
+                return distance;
+            }
+
+            distance -= itemWidths[i];
+        }
+
+        return distance;
+    }
+
+    /**
+     * Gets the maximum distance that can be scrolled
+     */
+    private getMaxScrollDistance(availableWidth: number, itemWidths: number[]): number {
+        const totalWidth: number = itemWidths.reduce((a: number, b: number) => a + b);
+        return totalWidth - availableWidth;
+    }
+
+    /**
+     * Handler for the previous click event
+     */
+    private handlePreviousClick = (): void => {
+        this.handleClick(ButtonDirection.previous);
+    }
+
+    /**
+     * Handler for the next click event
+     */
+    private handleNextClick = (): void => {
+        this.handleClick(ButtonDirection.next);
+    }
+
+    /**
+     * Handler for the click event
+     */
+    private handleClick(direction: ButtonDirection): void {
+        if (!canUseDOM()) {
+            return;
+        }
+
+        const availableWidth: number = getClientRectWithMargin(this.horizontalOverflowItemsRef.current).width;
+        const items: HTMLElement[] = Array.prototype.slice.call(this.horizontalOverflowItemsRef.current.childNodes);
+        const itemWidths: number[] = [];
+
+        for (const item of items) {
+            itemWidths.push(getClientRectWithMargin(item).width);
+        }
+
+        this.setScrollDistance(
+            this.getScrollDistanceFromDirection(
+                direction,
+                availableWidth,
+                itemWidths,
+                this.horizontalOverflowItemsRef.current.scrollLeft
+            )
+        );
+    }
+
+    /**
+     * Sets the scroll distance for the items list
+     */
+    private setScrollDistance(updatedDistance: number): void {
+        this.scrollLeft(
+            this.horizontalOverflowItemsRef.current,
+            updatedDistance,
+            this.props.scrollDuration ? this.props.scrollDuration : 500
+        );
+    }
+
+    /**
+     * Easing animation
+     * Inspired by the github gist contribution: https://gist.github.com/andjosh/6764939
+     */
+    private easeInOutQuad(
+        currentTime: number,
+        startValue: number,
+        changeInValue: number,
+        duration: number
+    ): number {
+        currentTime /= duration / 2;
+
+        if (currentTime < 1) {
+            return changeInValue / 2 * currentTime * currentTime + startValue;
+        }
+
+        currentTime--;
+
+        return -changeInValue / 2 * (currentTime * (currentTime - 2) - 1) + startValue;
+    }
+
+    /**
+     * Scrolls the container for the items list
+     */
+    private scrollLeft(element: HTMLUListElement, left: number, duration: number): void {
```

Comment:
Do we need to calculate this differently to account for localization? When testing this, next didn't operate as expected, but the previous button was required to fire when moving to RTL.

From the MDN docs:
```
Note that if the element's direction of the element is rtl (right-to-left) then scrollLeft is 0 when the scrollbar is at its rightmost position (at start of the scrolled content) and then increasingly negative as you scroll towards the end of the content.
```

Reply:
Yep, good catch

Reply:
see change on line 184